x-pack/metricbeat/module/prometheus/remote_write/data.go

```diff
@@ -0,0 +1,272 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package remote_write
+
+import (
+	"math"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+
+	dto "github.com/prometheus/client_model/go"
+	"github.com/prometheus/common/model"
+
+	"github.com/elastic/beats/v7/libbeat/common"
+	"github.com/elastic/beats/v7/libbeat/common/cfgwarn"
+	"github.com/elastic/beats/v7/libbeat/logp"
+	p "github.com/elastic/beats/v7/metricbeat/helper/prometheus"
+	"github.com/elastic/beats/v7/metricbeat/mb"
+	"github.com/elastic/beats/v7/metricbeat/module/prometheus/remote_write"
+	"github.com/elastic/beats/v7/x-pack/metricbeat/module/prometheus/collector"
+)
+
+const (
+	counterType   = "counter_type"
+	histogramType = "histgram_type"
+	otherType     = "other_type"
```

Comment:
We handle like gauges the summaries too so this why I tried to avoid to name this `gauge` explicitly in order to avoid the confusion of 1v1 mapping. However, do you think we should change it?

Reply:
Oh ok, it is not only about gauges :+1: Should we mention in some place that summaries are handled as gauges?